Extract ffmpeg to a separate function and add output logs if conversion fails

arsaboo opened the pull request:
Following up on our discussion in #9, I extracted the ffmpeg conversion process in a separate function and added logging if conversion fails. 

Let me know what you think.

Liupold commented on fmdpy/download.py:
```diff
@@ -1,14 +1,25 @@
+from fmdpy import config, headers, utils
+
+
+def convert_audio_to_mp3(input_file_path, output_file_path, bitrate):
```
This will fail for '.opus' output file names. (Probably because of the codec option!)

arsaboo replied:
Converted to a generic function.

Liupold commented:
good work on the log functionality.  But the codec option will create problem. If some other format is specified then this would cause the output file extension will be OK but the data format will NOT be OK. I would appreciate if you can write a general function which implements this. Something Like `ffmpeg_convert`.

arsaboo commented:
Moved to `pydub` and converted it to a generic function. I tested it with both mp3 and opus and it works fine. The tests are failing for unrelated reasons.

arsaboo commented:
Also, if we remove `ffmpeg`, we can remove 
https://github.com/Liupold/fmdpy/blob/e7492a208cb39894d7986d21208e250b98e1e0c1/.github/workflows/integrate.yaml#L19

Looks like that is causing the failed tests.

Liupold commented:
Are you ready to merge ? I have made a shadow branch in which the CI tests is  working. The CI test is failing because of the lack of the ffmpeg API key in your repo.

Liupold commented:
If you do add more commits, pls change the target branch from `main` to `debug`.  So your code will be merging into my debug branch and then I will push it to main. BTW, the conversion is working perfectly fine. Good  job!

arsaboo commented:
I'm ready to merge...I tested this for a while and it's working well for me for various formats. Feel free to merge this. Thanks!!
